### PR TITLE
Fix EZP-23985: Refactor: FieldNameGenerator should be shared by all search engines

### PR DESCRIPTION
**Pull request by pspanja**
This PR resolves https://jira.ez.no/browse/EZP-23985, sub-task of https://jira.ez.no/browse/EZP-23940

Followup: https://jira.ez.no/browse/EZP-23991

Common `FieldNameGenerator` is implemented and container is configured with field mappings for Elasticsearch and Solr Search Engines. Old implementations are removed.

Symfony intgration is manually tested.

### CODE REVIEW

**Comment by pspanja** [comment]
Ready for review, ping @andrerom, @bdunogier, @glye, @lolautruche, @yannickroger.

File: eZ/Publish/Core/settings/search_engines/common.yml [hunk 1]
```diff
@@ -1,8 +1,32 @@
...
+                ez_html: 'h'
+                ez_boolean: 'b'
+                ez_mboolean: 'mb'
+                ez_float: 'f'
+                ez_double: 'd'
+                ez_date: 'dt'
+                ez_point: 'p'
+                ez_currency: 'c'
+                ez_geolocation: 'gl'
+                ez_document: 'doc'
```
Comment (lolautruche) [inline eZ/Publish/Core/settings/search_engines/common.yml]: Please add a parameter for that.

**Comment by lolautruche** [comment]
Apart my remark, +1

File: eZ/Publish/Core/settings/search_engines/common.yml (hunk 1 above)
Comment (pspanja) [inline eZ/Publish/Core/settings/search_engines/common.yml]: Thanks, fixed and rebased.

**Comment by andrerom** [comment]
+1 if travis agrees

**Comment by pspanja** [comment]
Thanks!

**Pull request merged**